File: 504/Solution.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"
        x = []
        neg = False
        temp = abs(num)
        while temp:
            x.append(str(temp%7))
            temp = temp//7
        
        return "-"*(num<0) + "".join(x[::-1])

File: 504/Solution.java
import java.lang.Math;

class Solution {
    public String convertToBase7(int num) {
        if(num == 0){
            return "0";
        }
        StringBuilder sb = new StringBuilder();
        int temp = Math.abs(num);
        while(temp > 0) {
            sb.append(temp%7);
            temp = temp/7;
        }
        if(num<0)sb.append("-");
        return sb.reverse().toString();
    }
}